File: hyejin/Week_15/Rotate Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        // 실질적으로 nums.length mod k 만큼만 이동하면 됨
		k %= nums.length;

		int[] copy = nums.clone();
		for (int curIndex = 0; curIndex < nums.length; curIndex++) {
			int newIndex = curIndex + k;
			if (newIndex >= nums.length)
				newIndex -= nums.length;
			nums[newIndex] = copy[curIndex];
		}
    }
}

File: sangryeol/Week_15/Rotate Array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int> copyNums = nums;
        int N = nums.size();
        for (int i = 0; i < N; i++) {
            nums[(i + k) % N] = copyNums[i];
        }
        return;
    }
};